src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/PreventNumericIntPtrUIntPtrBehavioralChanges.cs

```diff
@@ -0,0 +1,183 @@
+            CreateLocalizableResourceString(nameof(PreventNumericIntPtrUIntPtrBehavioralChangesOperatorNotThrowMessage)),
+            DiagnosticCategory.Reliability,
+            RuleLevel.BuildWarning,
+            CreateLocalizableResourceString(nameof(PreventNumericIntPtrUIntPtrBehavioralChangesDescription)),
+            isPortedFxCopRule: false,
+            isDataflowRule: false);
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; } = ImmutableArray.Create(OperatorThrowsRule, NotThrowRule);
+
+        protected abstract bool IsWithinCheckedContext(IOperation operation);
+
+        protected abstract bool IsAliasUsed(ISymbol? syntaxReferences);
+
+        protected abstract bool IsAliasUsed(SyntaxNode syntax);
+
+        public override void Initialize(AnalysisContext context)
+        {
+            context.EnableConcurrentExecution();
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+            context.RegisterCompilationStartAction(context =>
+            {
+                if (!context.Compilation.TryGetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemRuntimeCompilerServicesRuntimeFeature, out var runtimeFeatureType) ||
+                    !runtimeFeatureType.GetMembers("NumericIntPtr").OfType<IFieldSymbol>().Any())
+                {
+                    // Numeric IntPtr feature not available
+                    return;
+                }
+
+                context.RegisterOperationAction(context =>
+                {
+                    if (context.Operation is IBinaryOperation binaryOperation &&
+                        binaryOperation.IsAdditionOrSubstractionOperation(out var binaryOperator) &&
+                        binaryOperation.IsChecked)
+                    {
+                        if ((binaryOperation.LeftOperand.Type?.SpecialType == SpecialType.System_IntPtr ||
+                             binaryOperation.LeftOperand.Type?.SpecialType == SpecialType.System_UIntPtr) &&
+                             IsConversionFromInt32(binaryOperation.RightOperand) &&
+                             !IsAliasUsed(GetSymbol(binaryOperation.LeftOperand)))
+                        {
+                            context.ReportDiagnostic(binaryOperation.CreateDiagnostic(OperatorThrowsRule, binaryOperator));
+                        }
+                    }
+                    else if (context.Operation is IConversionOperation conversionOperation)
+                    {
+                        var operation = conversionOperation.WalkDownConversion(c => c.IsImplicit); // get innermost converesion
+                        if (operation is IConversionOperation explicitConversion &&
+                            explicitConversion.OperatorMethod == null) // Built in conversion
+                        {
+                            if (IsWithinCheckedContext(explicitConversion)) // explicitConversion.IsChecked somehow not working
```

Comment:
> // explicitConversion.IsChecked somehow not working

Is it not working (warranting an issue being filed) or is this defensive code (perhaps warranting a `Debug.Assert`)?

Reply:
`IConversionOperation.IsChecked` was not working (was false in checked context), whereas `IBinalryOperation.IsChecked` was working, so it might be a bug, commented here as I was not sure. 

CC @mavasani  @jmarolf

Reply:
file a [bug](https://github.com/dotnet/roslyn/issues/63951)